src/Security/Filter/TenantFilter.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace RL\Security\Filter;
+
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\Query\Filter\SQLFilter;
+use Doctrine\Common\Annotations\Reader;
+
+class TenantFilter extends SQLFilter
+{
+    /** @var Reader */
+    protected Reader $reader;
+
+    /**
+     * @param ClassMetadata $targetEntity
+     * @param string $targetTableAlias
+     * @return string
+     */
+    public function addFilterConstraint(ClassMetadata $targetEntity, $targetTableAlias)
+    {
+        $query = '';
+
+        if (empty($this->reader)) {
+            return '';
+        }
+
+        if ($this->hasParameter('tenant')) {
+            $fieldName = 'tenant';
+        } else {
+            $fieldName = 'app_id';
```

Comment:
should probably add a check for this field as well instead of assuming it's always there if `tenant` isnt